bridges/zap/zap_logger.go

```diff
@@ -0,0 +1,119 @@
+package zap
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/log/noop"
+	"go.uber.org/zap/zapcore"
+)
+
+const (
+	bridgeName = "go.opentelemetry.io/contrib/bridge/zapcore"
+)
+
+type OtelZapCore struct {
+	logger log.Logger
+	attr   []log.KeyValue
+}
+
+var (
+	_ zapcore.Core = (*OtelZapCore)(nil)
+)
+
+// this function creates a new zapcore.Core that can be used with zap.New()
+// this instance will translate zap logs to opentelemetry logs and export them
+func NewOtelZapCore(lp log.LoggerProvider, opts ...log.LoggerOption) zapcore.Core {
```

Comment:
This needs to accept its own package defined `Option` type.

https://github.com/open-telemetry/opentelemetry-go-contrib/blob/851b2fd97fa45bb502cc094b6e3e84464a140fc3/bridges/otelslog/version.go#L7